data-prepper-plugins/elasticsearch/src/main/java/com/amazon/dataprepper/plugins/sink/elasticsearch/ConnectionConfiguration.java

```diff
@@ -338,6 +339,8 @@ public Builder withAwsRegion(final String awsRegion) {
     }
 
     public Builder withAWSStsRoleArn(final String awsStsRoleArn) {
+      checkArgument(awsStsRoleArn == null || awsStsRoleArn.length() <= 2048, "awsStsRoleArn length cannot exceed 2048");
+      checkArgument(awsStsRoleArn == null || awsStsRoleArn.equals(Arn.fromString(awsStsRoleArn).toString()), "Invalid ARN format for awsStsRoleArn");
```

Comment:
I think for malformatted `awsStsRoleArn`, Arn.fromString(awsStsRoleArn) will fail immediately and as a result it will not reach the custom checkArgument message. Maybe we could do

```
try {
  Arn.fromString(awsStsRoleArn);
} catch (Exception e) {
   throw new IllegalArgumentException("Invalid ARN format for awsStsRoleArn")
}
```